src/modules/s3.js

```diff
@@ -0,0 +1,98 @@
+import {
+  S3Client,
+  ListBucketsCommand,
+  ListObjectsCommand,
+  PutObjectCommand, GetObjectCommand, DeleteObjectCommand,
+} from '@aws-sdk/client-s3';
+import dotenv from 'dotenv';
+import assert from 'assert';
+import fs from 'fs';
+import path from 'path';
+import { __dirname } from './globals.js';
+
+const REGION = 'us-east-1';
+dotenv.config();
+
+// Validate whether env vars are set or not. (Fail-fast)
+if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+  throw new Error(`
+       You haven't set the AWS_SECRET_ACCESS_KEY or
+       the AWS_ACCESS_KEY_ID variable(s) in your env file.
+    `);
+}
+
+const s3 = new S3Client({ region: REGION });
+
+/**
+ * Fetches the all of the AWS S3 buckets.
+ *
+ * @returns {Promise<Bucket[]>}
+ */
+const getBuckets = async () => s3.send(new ListBucketsCommand({}))
+  .then(result => result.Buckets);
+
+/**
+ * @param param
+ * @param {String} param.Bucket The name of the bucket
+ * @returns {Promise<Object[]>}
+ */
+const listObjects = async (param = {
+  Bucket: 'image-sharing-project',
```

Comment:
Should be `Bucket: param.Bucket || 'image-sharing-project'`,

Reply:
`param` is the parameter itself, if we do `Bucket: param.Bucket || 'image-sharing-project'` it's just going to reference itself.